src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -73,6 +73,8 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
                 builder.Insert(i++, @using);
             }
 
+            InheritImportedDirectives(document, codeDocument.Source.FilePath);
```

Comment:
Preserving the source ordering is **critical** 👍 If it makes our jobs harder, too bad, that's what we signed up for.